lib/components/segmented-control/segmented-control.scss

```diff
@@ -0,0 +1,154 @@
+@import "../../styles/index.scss";
+@import "../../styles/quill/border.scss";
+@import "../../styles/quill/color.scss";
+@import "../../styles/quill/color.semantic.scss";
+@import "../../styles/quill/motion.scss";
+@import "../../styles/quill/opacity.scss";
+@import "../../styles/quill/others.scss";
+@import "../../styles/quill/spacing.scss";
+@import "../../styles/quill/static.scss";
+
+.segmented-control {
+    &-single {
+        display: grid;
+        grid-auto-flow: column;
+        grid-auto-columns: 1fr;
+        width: fit-content;
+        background: color-mix(
+            in srgb,
+            var(--semantic-color-typography-prominent) 4%,
+            transparent
+        );
+
+        .item {
+            display: flex;
+            justify-content: center;
+            align-items: center;
+            cursor: pointer;
+            position: relative;
+            z-index: 2;
+
+            * {
+                z-index: 2;
+            }
+            &:disabled {
+                opacity: var(--core-opacity-300);
+                cursor: not-allowed;
+                pointer-events: none;
+            }
+            &.focused,
+            &.focused::before {
+                // not using default :focus since we are customizing focus state
```

Comment:
You still can use natural pseudo-class, this will help to get rid of all logic in ._tsx_, where you used `useState` `focused`.
But, you need NOT `:focus`, but `:focus-visible` as it will not apply focus your button after click. If you want to customize, you can remove original style by adding `outline: none;` and add your own

Reply:
unfortunately, i can't get rid of useState `focused` because i need to completely customize focus behaviour, not only after click but also for pressing different keys. For example, pressing "Tab" and "Left/Right" arrow keys should have different behavior, even though they both make buttons be focused.

Reply:
Sure, you could ignore my comment if it wasn't relevant)